keep/api/routes/preset.py

```diff
@@ -119,6 +119,38 @@ def pull_data_from_providers(
                 extra=extra,
             )
 
+            if isinstance(provider_class, BaseIncidentProvider):
+                try:
+                    incidents = provider_class.get_incidents()
+                    process_event(
+                        {},
+                        tenant_id,
+                        provider.type,
+                        provider.id,
+                        None,
+                        None,
+                        trace_id,
+                        incidents,
+                        notify_client=False,
+                        timestamp_forced=None,
+                        process_event_as="incident"
+                    )
+                except NotImplementedError:
+                    logger.debug(
+                        f"Provider {provider.type} ({provider.id}) does not implement pulling incidents",
+                        extra=extra,
+                    )
+                except Exception as e:
+                    logger.error(
+                        f"Unknown error pulling incidents from provider {provider.type} ({provider.id})",
+                        extra={**extra, "error": str(e)},
+                    )
+            else:
+                logger.debug(
+                    f"Provider {provider.type} ({provider.id}) does not implement pulling incidents",
+                    extra=extra,
+                )
```

Comment:
Again, this is something we want to refactor, maybe add an async "pull_incidents" when a user views the incident page?